src/codeu/chat/DerbyStore.java

```diff
@@ -81,18 +82,18 @@ public DerbyStore() {
 			stmt = conn.createStatement();
 			
 			// Create the chat user table
-			stmt.execute("CREATE TABLE " + userTableName + "(id varchar(255), name varchar(255), password varchar(255), creation BIGINT)");
+			stmt.execute("CREATE TABLE " + userTableName + " (userid varchar(255), name varchar(255), password varchar(255), creation BIGINT, PRIMARY KEY (userid))");
```

Comment:
Yep, you can absolutely have two primary keys. (You can also have a primary key that's based on multiple columns, but that's not what you want here.) Then you'd probably want to add more robust handling of the case where a user tries to register with a name that's already used.